repository/kafka/operator/operator.yaml

```diff
@@ -34,38 +48,35 @@ tasks:
         - file: /tmp/tls.crt
           kind: Secret
           key: certificate
-  - name: user-workload
-    kind: Apply
+  # feature tasks
+  - name: service-monitor
+    kind: Toggle
     spec:
+      parameter: ADD_SERVICE_MONITOR
       resources:
-      - user-workload.yaml
-  - name: service
-    kind: Apply
+        - service-monitor.yaml
+  - name: external-access
+    kind: Toggle
     spec:
+      parameter: EXTERNAL_ADVERTISED_LISTENER
       resources:
-      - service.yaml
-      - service-monitor.yaml
-  - name: external-access
-    kind: Apply
+        - clusterrole.yaml
+        - clusterrolebinding.yaml
+        - external-service.yaml
+  - name: mirrormaker
+    kind: Toggle
     spec:
+      parameter: MIRROR_MAKER_ENABLED
       resources:
-      - clusterrole.yaml
-      - clusterrolebinding.yaml
-      - external-service.yaml
-  - name: configuration
-    kind: Apply
+        - mirror-maker-config.yaml
+        - mirror-maker.yaml
+  - name: user-workload
+    kind: Toggle
     spec:
+      parameter: RUN_USER_WORKLOAD
       resources:
-      - serviceaccount.yaml
-      - rolebinding.yaml
-      - role.yaml
-      - jaas-config.yaml
-      - krb5-config.yaml
-      - server.properties.yaml
-      - bootstrap.yaml
-      - metrics-config.yaml
-      - health-check.yaml
-      - enable-tls.yaml
+        - user-workload.yaml
+  # task used to restrict patching sts where it is not allowed
   - name: not-allowed
     kind: Dummy
```

Comment:
On the topic of the `not-allowed` task: Would it make sense to have a task type "Fail" that simply FATAL_ERRORs the plan with a specified message?

Reply:
yeah we need to do another round here now after many releases

Reply:
Actually, I think we really need `immutable` parameters. Having the task is ok-ish for now, but the parameters still get set even if the plan fails. And on the next run of `update` they're applied

Reply:
And our admission controller would be able to prevent changing immutable parameters. We need a KEP for that but otherwise, the solution can be pretty simple.